src/native/managed/cdacreader/src/Target.cs

```diff
@@ -16,49 +17,156 @@ public struct TargetPointer
 
 internal sealed unsafe class Target
 {
+    private const int StackAllocByteThreshold = 1024;
+
     private readonly delegate* unmanaged<ulong, byte*, uint, void*, int> _readFromTarget;
     private readonly void* _readContext;
 
     private bool _isLittleEndian;
     private int _pointerSize;
 
-    public Target(ulong _, delegate* unmanaged<ulong, byte*, uint, void*, int> readFromTarget, void* readContext)
+    private TargetPointer[] _pointerData = [];
+    private IReadOnlyDictionary<string, int> _contracts = new Dictionary<string, int>();
+
+    public Target(ulong contractDescriptor, delegate* unmanaged<ulong, byte*, uint, void*, int> readFromTarget, void* readContext)
     {
         _readFromTarget = readFromTarget;
         _readContext = readContext;
 
-        // TODO: [cdac] Populate from descriptor
-        _isLittleEndian = BitConverter.IsLittleEndian;
-        _pointerSize = IntPtr.Size;
+        ReadContractDescriptor(contractDescriptor);
```

Comment:
I think it's just personal preference, but I always find constructors that call helper methods that are responsible for initializing some members harder to think about ("which helper methods call I call at this point? do they use members that haven't been inited yet?").

Also, instead of a throwing constructor, I tend to prefer a static builder method (that can return false or a detailed diagnostic) that does most of the work upfront and then calls a `private` constructor that just assigns some members.